projection/projections.go

```diff
@@ -0,0 +1,51 @@
+package projection
+
+import (
+	"fmt"
+
+	"github.com/crypto-com/chain-indexing/appinterface/cosmosapp"
+	"github.com/crypto-com/chain-indexing/appinterface/rdb"
+	projection_entity "github.com/crypto-com/chain-indexing/entity/projection"
+	applogger "github.com/crypto-com/chain-indexing/internal/logger"
+	"github.com/crypto-com/chain-indexing/projection/account"
+	"github.com/crypto-com/chain-indexing/projection/account_message"
+	"github.com/crypto-com/chain-indexing/projection/block"
+	"github.com/crypto-com/chain-indexing/projection/blockevent"
+	"github.com/crypto-com/chain-indexing/projection/transaction"
+	"github.com/crypto-com/chain-indexing/projection/validator"
+	"github.com/crypto-com/chain-indexing/projection/validatorstats"
+)
+
+func InitProjection(name string, params InitParams) projection_entity.Projection {
+	switch name {
+	case "Account":
+		return account.NewAccount(params.Logger, params.RdbConn, params.CosmosAppClient, params.BaseDenom)
+	case "AccountMessage":
```

Comment:
Is it any benefit to extract cases MATCHERS out to a constant file?

Reply:
Um, could do. As long as the constant and init is in the same file. The design is to make adding new projection does not need to amend too many files (and may miss some if too many)